File: ui/src/main/java/com/midtrans/sdk/corekit/core/themes/CustomColorTheme.java
package com.midtrans.sdk.corekit.core.themes;

import android.graphics.Color;


/**
 * Created by rakawm on 2/16/17.
 */

public class CustomColorTheme implements BaseColorTheme {
    private String colorPrimaryHex;
    private String colorPrimaryDarkHex;
    private String colorSecondaryHex;

    public CustomColorTheme(String colorPrimaryHex, String colorPrimaryDarkHex, String colorSecondaryHex) {
        this.colorPrimaryHex = colorPrimaryHex;
        this.colorPrimaryDarkHex = colorPrimaryDarkHex;
        this.colorSecondaryHex = colorSecondaryHex;
    }

    @Override
    public int getPrimaryColor() {
        try {
            if (!colorPrimaryHex.startsWith("#")) {
                return Color.parseColor("#" + colorPrimaryHex.toLowerCase());
            }

            return Color.parseColor(colorPrimaryHex.toLowerCase());
        } catch (Exception exception) {
            return Color.parseColor("#999999");
        }
    }

    @Override
    public int getPrimaryDarkColor() {
        try {
            if (!colorPrimaryDarkHex.startsWith("#")) {
                return Color.parseColor("#" + colorPrimaryDarkHex.toLowerCase());
            }

            return Color.parseColor(colorPrimaryDarkHex.toLowerCase());
        } catch (Exception exception) {
            return Color.parseColor("#737373");
        }
    }

    @Override
    public int getSecondaryColor() {
        try {
            if (!colorSecondaryHex.startsWith("#")) {
                return Color.parseColor("#" + colorSecondaryHex.toLowerCase());
            }

            return Color.parseColor(colorSecondaryHex.toLowerCase());
        } catch (Exception exception) {
            return Color.parseColor("#adadad");
        }
    }
}

File: ui/src/main/kotlin/com/midtrans/sdk/uikit/api/model/CustomColorTheme.kt
package com.midtrans.sdk.uikit.api.model

import android.graphics.Color
import java.util.*

data class CustomColorTheme(
    val colorPrimaryHex: String? = null,
    val colorPrimaryDarkHex: String? = null,
    val colorSecondaryHex: String? = null,
) {
    fun getPrimaryColor(): Int {
        return try {
            if (!colorPrimaryHex!!.startsWith("#")) {
                Color.parseColor("#" + colorPrimaryHex.lowercase(Locale.getDefault()))
            } else Color.parseColor(colorPrimaryHex.lowercase(Locale.getDefault()))
        } catch (exception: Exception) {
            Color.parseColor("#999999")
        }
    }

    fun getPrimaryDarkColor(): Int {
        return try {
            if (!colorPrimaryDarkHex!!.startsWith("#")) {
                Color.parseColor("#" + colorPrimaryDarkHex.lowercase(Locale.getDefault()))
            } else Color.parseColor(colorPrimaryDarkHex.lowercase(Locale.getDefault()))
        } catch (exception: Exception) {
            Color.parseColor("#737373")
        }
    }

    fun getSecondaryColor(): Int {
        return try {
            if (!colorSecondaryHex!!.startsWith("#")) {
                Color.parseColor("#" + colorSecondaryHex.lowercase(Locale.getDefault()))
            } else Color.parseColor(colorSecondaryHex.lowercase(Locale.getDefault()))
        } catch (exception: Exception) {
            Color.parseColor("#adadad")
        }
    }
}